app/src/test/java/com/github/onlynotesswent/model/note/NoteRepositoryFirestoreTest.kt

```diff
@@ -349,4 +350,36 @@ class NoteRepositoryFirestoreTest {
 
     verify(timeout(100)) { (mockQuerySnapshot).documents }
   }
+
+  @Test
+  fun deleteNotesFromFolder_callsDocuments() {
+    `when`(mockDocumentReference.delete()).thenReturn(Tasks.forResult(null))
+
+    noteRepositoryFirestore.deleteNotesFromFolder("1", onSuccess = {}, onFailure = {})
+
+    shadowOf(Looper.getMainLooper()).idle()
+
+    // Ensure the delete method was called twice (once for each sub note)
+    verify(mockDocumentReference, times(2)).delete()
+  }
+
+  @Test
+  fun deleteNotesFromFolder_fails() {
+    val errorMessage = "TestError"
+    `when`(mockQuerySnapshotTask.isSuccessful).thenReturn(false)
+    `when`(mockQuerySnapshotTask.exception).thenReturn(Exception(errorMessage))
+    `when`(mockQuerySnapshotTask.addOnCompleteListener(any())).thenAnswer { invocation ->
+      val listener = invocation.getArgument<OnCompleteListener<QuerySnapshot>>(0)
+      // Simulate a result being passed to the listener
+      listener.onComplete(mockQuerySnapshotTask)
+      mockQuerySnapshotTask
+    }
+    `when`(mockQuerySnapshot.documents)
+        .thenReturn(listOf(mockDocumentSnapshot3, mockDocumentSnapshot4))
+    var exceptionThrown: Exception? = null
+    noteRepositoryFirestore.deleteNotesFromFolder(
+        "1", onSuccess = {}, onFailure = { e -> exceptionThrown = e })
+    assertNotNull(exceptionThrown)
```

Comment:
Nice, though you could also create a val with the test exception, and pass it to onFailure, then check if it is equal to the one that is called on mocked repository. Either way is equivalent though I think

Reply:
It is probably equivalent but I did it like this to make it similar to previously added failure tests